src/software/jetson_nano/services/motor.cpp

```diff
@@ -210,106 +208,188 @@ bool MotorService::checkDriverFault(uint8_t motor)
+            {
+                *(motor_status.mutable_front_left()) = drive_status;
+            }
+            if (motor == FRONT_RIGHT_MOTOR_CHIP_SELECT)
+            {
+                *(motor_status.mutable_front_right()) = drive_status;
+            }
+            if (motor == BACK_LEFT_MOTOR_CHIP_SELECT)
+            {
+                *(motor_status.mutable_back_left()) = drive_status;
+            }
+            if (motor == BACK_RIGHT_MOTOR_CHIP_SELECT)
+            {
+                *(motor_status.mutable_back_right()) = drive_status;
+            }
+        }
+        else
+        {
+            TbotsProto::DribblerStatus dribbler_status;
+            dribbler_status.set_dribbler_rpm(static_cast<float>(dribbler_rpm));
+            dribbler_status.set_enabled(cached_motor_faults_[motor].drive_enabled);
+            for (const TbotsProto::MotorFault& fault :
+                 cached_motor_faults_[motor].motor_faults)
+            {
+                dribbler_status.add_motor_faults(fault);
+            }
+
+            *(motor_status.mutable_dribbler()) = dribbler_status;
+        }
+    }
+
+    motor_status.mutable_front_left()->set_wheel_velocity(
+        static_cast<float>(front_left_velocity_mps));
+    motor_status.mutable_front_right()->set_wheel_velocity(
+        static_cast<float>(front_right_velocity_mps));
+    motor_status.mutable_back_left()->set_wheel_velocity(
+        static_cast<float>(back_left_velocity_mps));
+    motor_status.mutable_back_right()->set_wheel_velocity(
+        static_cast<float>(back_right_velocity_mps));
+
+    motor_fault_detector_ =
+        static_cast<uint8_t>((motor_fault_detector_ + 1) % NUM_MOTORS);
+
+    return motor_status;
+}
 
 TbotsProto::MotorStatus MotorService::poll(const TbotsProto::MotorControl& motor,
                                            double time_elapsed_since_last_poll_s)
 {
-    TbotsProto::MotorStatus motor_status;
-
     bool encoders_calibrated = (encoder_calibrated_[FRONT_LEFT_MOTOR_CHIP_SELECT] ||
                                 encoder_calibrated_[FRONT_RIGHT_MOTOR_CHIP_SELECT] ||
                                 encoder_calibrated_[BACK_LEFT_MOTOR_CHIP_SELECT] ||
                                 encoder_calibrated_[BACK_RIGHT_MOTOR_CHIP_SELECT]);
 
-    int reset_detector = tmc4671_readInt(0, TMC4671_PID_ACCELERATION_LIMIT);
-
-    // When the motor board is reset the value in the above register is set to the maximum
-    // signed 32 bit value Please read the header file and the datasheet for more info
-    if (reset_detector == 2147483647)
+    if (!encoders_calibrated)
     {
-        LOG(DEBUG) << "RESET DETECTED";
-        setUpMotors();
-        encoders_calibrated = false;
+        is_initialized_ = false;
     }
-    // check if encoders are calibrated
-    if (!encoders_calibrated)
+
+    // checks if any motor has reset, sends a log message if so
+    for (uint8_t motor = 0; motor < NUM_MOTORS; ++motor)
```

Comment:
nima: Does encoder calibration and checking encoder_calibrated_ have to be done in poll? Can't it be done in setup? Since if I understand the logic correctly, we only calibrate at initialization and if a reset was detected.

Reply:
I moved the calibration to setup() @nimazareian